DataVaildation/DataVaildation.robot

```diff
@@ -0,0 +1,131 @@
+    [Teardown]    Disconnect From Database
+
+Verify ColumnDataTypes
+    [Documentation]    Verify the data types of each column.
+    [Setup]    Connect to DB
+    ${data_types}    Query    SHOW COLUMNS FROM tsao_accounts
+
+    FOR    ${column_info}    IN    @{data_types}
+        ${column_data_type}    Set Variable    ${column_info[1]}
+        Log    Column Data Type: ${column_data_type}
+        Run Keyword    Verify Data Type Contains    ${column_data_type.lower()}    int
+        Run Keyword    Verify Data Type Contains    ${column_data_type.lower()}    varchar(100)
+        Run Keyword    Verify Data Type Contains    ${column_data_type.lower()}    varchar(100)
+        Run Keyword    Verify Data Type Contains    ${column_data_type.lower()}    char(64)
+        Run Keyword    Verify Data Type Contains    ${column_data_type.lower()}    enum('Administrator', 'User')
+        Run Keyword    Verify Data Type Contains    ${column_data_type.lower()}    datetime
+        Run Keyword    Verify Data Type Contains    ${column_data_type.lower()}    bool
+        Run Keyword    Verify Data Type Contains    ${column_data_type.lower()}    bool
+    END
+
+    [Teardown]    Disconnect From Database
+Verify UniqueConstraint
+    [Documentation]    Verify the unique constraint on the 'Username' column.
+    [Setup]    Connect to DB
+    ${unique_constraint}    Query    SELECT COUNT(*) FROM INFORMATION_SCHEMA.STATISTICS WHERE table_name = 'tsao_accounts' AND column_name = 'Username' AND non_unique = 0
+
+    Run Keyword If    ${unique_constraint[0][0]} > 0    Log    Unique constraint exists for the 'Username' column
+    Run Keyword If    ${unique_constraint[0][0]} == 0    Fail    No unique constraint found for the 'Username' column
+
+    [Teardown]    Disconnect From Database
+Verify EnumValues
+    [Documentation]    Verify that the 'Role' column only accepts values 'Administrator' and 'User'.
+    [Setup]    Connect to DB
+    ${enum_values}    Query    SELECT DISTINCT Role FROM tsao_accounts
+    List Should Contain Value    ${enum_values[0]}    Administrator
+    List Should Contain Value    ${enum_values[1]}    User
+    [Teardown]    Disconnect From Database
+
+Verify DefaultValues
+    [Documentation]    Verify default values for 'IsApproved' and 'IsDeleted'.
+    [Setup]    Connect to DB
+    ${default_values}    Query    SELECT IsApproved, IsDeleted FROM tsao_accounts WHERE ID=1
+
+    Run Keyword If    ${default_values}    Verify IsApproved and IsDeleted Default Values    ${default_values}
+    Run Keyword If    not ${default_values}    Fail    No default values found for the 'IsApproved' and 'IsDeleted' columns
+
+    [Teardown]    Disconnect From Database
+
+Check If Exists In DB
+    [Documentation]    Verify that specific data exist in database.
+    [Setup]    Connect to DB
+    Check If Exists In Database    SELECT ID FROM tsao_accounts WHERE Name = 'Ben Low';
+    [Teardown]    Disconnect From Database
+
+Check If Not Exists In DB
```

Comment:
maybe this test title can be more specific, since it is checking if there is a row with the ContactRole 'Lecturer'